src/view/com/post-thread/PostThread.tsx

```diff
@@ -163,42 +169,75 @@ function PostThreadLoaded({
     [threadViewPrefs, thread],
   )
 
-  // construct content
+  // On web we always are ready to render, we only need this on mobile.
+  const [readyToShowAll, setReadyToShowAll] = React.useState(isWeb)
+  const [topPageCount, setTopPageCount] = React.useState(1)
+  const isPrepending = React.useRef(false)
+
```

Comment:
`readyToShowAll` gets set to true after we have rendered the first item. We need to wait a moment before setting this to true, hence the 300ms `setTimeout`.

`topPageCount` is just us keeping track of how many pages we have displayed at the top (15 per page)

`isPrepending` ensures we do not run the `onStartReached` callback too many times.